code/go/0chain.net/blobbercore/allocation/repository.go

```diff
@@ -149,16 +187,76 @@ func (r *Repository) UpdateAllocationRedeem(ctx context.Context, allocationID, A
 	if err != nil {
 		return err
 	}
-	delete(cache, allocationID)
 
 	allocationUpdates := make(map[string]interface{})
 	allocationUpdates["latest_redeemed_write_marker"] = AllocationRoot
 	allocationUpdates["is_redeem_required"] = false
 	err = tx.Model(allocationObj).Updates(allocationUpdates).Error
-	return err
+	if err != nil {
+		return err
+	}
+	allocationObj.LatestRedeemedWM = AllocationRoot
```

Comment:
can there be a concurrent issue when this update will be run in different threads?

Reply:
We lock the allocation before this update by using `GetByIdAndLock` so this update will be synchronous.